### PR TITLE
ProjectView files parser implementation

### PR DESCRIPTION
**Pull request by Abrams27**
I forgot a bit how to code in java, but hopefully it works.

This is an early stage implementation of our ProjectView files parser.
Hopefully, the code structure will allow to add support for new fields easy.

TODO:
- `import` support
- use projectview on the server side

### CODE REVIEW

File: src/main/java/org/jetbrains/bsp/bazel/projectview/parser/splitter/ProjectViewSectionSplitter.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            .map(ProjectViewSectionHeaderPosition::getStartIndex)
+            .map(startIndex -> fileContent.substring(currentPosition.getEndIndex(), startIndex))
+            .orElse(fileContent.substring(currentPosition.getEndIndex()));
+    String sectionHeader = removeEndingHeaderColonAndSpaces(currentPosition.getHeader());
+
+    return new ProjectViewRawSection(sectionHeader, sectionBody);
+  }
+
+  private static String removeEndingHeaderColonAndSpaces(String header) {
+    return header.replace(":", "").trim();
```
Comment (andrefmrocha) [inline src/main/java/org/jetbrains/bsp/bazel/projectview/parser/splitter/ProjectViewSectionSplitter.java:88]: Maybe we can extract `:`  into a constant?

**Review by andrefmrocha — CHANGES_REQUESTED** [review]
Overall looks amazing, very good work!

Some concerns/overall questions I have:
- Are we going to merge this first without any connection to the server itself?
- Can we generate use files over string representation of the file contents in the tests?
- This logic is a bit tricky, it would be great if some minor comments could be added to help understanding it.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK 1.8
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: Run tests
+      run: bazel test //...
```
Comment (andrefmrocha) [inline .github/workflows/tests.yml:19]: Pls remind me to reproduce this in the teamcity build server

Comment (Abrams27) [reply .github/workflows/tests.yml:19]: currently it doesn't work, but i will fix it

**Comment by Abrams27** [comment]
> Overall looks amazing, very good work!
> 
> Some concerns/overall questions I have:
> 
> * Are we going to merge this first without any connection to the server itself?
Yea, i will introduce it in the following PR

> * Can we generate use files over string representation of the file contents in the tests?
Yea, sounds like a good idea

> * This logic is a bit tricky, it would be great if some minor comments could be added to help understanding it.
I tried to show it in the code structure, nevertheless will add some

File: src/test/java/org/jetbrains/bsp/bazel/projectview/parser/ProjectViewParserImplTest.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+import static org.junit.Assert.assertEquals;
+
+public class ProjectViewParserImplTest {
+
+  private ProjectViewParser parser;
+
+  @Before
+  public void before() {
+    this.parser = new ProjectViewParserImpl();
+  }
```
[the remark this answers is not in the document]
Comment (Abrams27) [reply src/test/java/org/jetbrains/bsp/bazel/projectview/parser/ProjectViewParserImplTest.java:22]: yea, so I had in my mind securing tests execution from possible side effects of the parser

**Comment by Abrams27** [comment]
Merging, any further fixes will be done in the following PR

File: src/test/java/org/jetbrains/bsp/bazel/projectview/parser/ProjectViewParserImplTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (andrefmrocha) [reply src/test/java/org/jetbrains/bsp/bazel/projectview/parser/ProjectViewParserImplTest.java:27]: What possible side effects are we talking about?

[the remark this answers is not in the document]
Comment (Abrams27) [reply src/test/java/org/jetbrains/bsp/bazel/projectview/parser/ProjectViewParserImplTest.java:27]: better safe than sorry, just protection against any future changes

**Comment by Abrams27** [comment]
merging, to unblock further changes

**Pull request merged**